src/fh-family-experience-web/Pages/FamilyHubResults.cshtml.cs

```diff
@@ -1,36 +1,52 @@
 namespace fh_family_experience_web.Pages;
 
-using fh_family_experience_web.Data;
-using fh_family_experience_web.Data.Entities;
 using fh_family_experience_web.Filters;
 using fh_family_experience_web.Services;
+using fh_family_experience_web.Services.Api;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Text.Json;
+using fh_family_experience_web.ViewModels;
+using fh_family_experience_web.ViewModels.Extensions;
+using fh_family_experience_web.Services.Postcodes.Model;
 
 [PageHistory]
 public class FamilyHubResultsModel : PageModel
 {
-    private readonly IReadRepository _repository;
     private readonly ILocalAuthorityLookupService _localAuthorityLookupService;
+    private readonly IServiceDirectoryApiClient _serviceDirectoryApi;
 
-    public FamilyHubResultsModel(IReadRepository repository, ILocalAuthorityLookupService localAuthorityLookupService)
+    public FamilyHubResultsModel(ILocalAuthorityLookupService localAuthorityLookupService, 
+        IServiceDirectoryApiClient serviceDirectoryApiClient)
     {
-        _repository = repository;
         _localAuthorityLookupService = localAuthorityLookupService;
+        _serviceDirectoryApi = serviceDirectoryApiClient;
+
+        FamilyHubs = new List<FamilyHubViewModel>();
     }
 
     [TempData]
     public string LocalAuthority { get; set; } = "";
 
     [TempData]
-    public string PostCode { get; set; } = "";
+    public string Postcode { get; set; } = "";
 
-    public List<Service> FamilyHubs { get; set; }
+    public IList<FamilyHubViewModel> FamilyHubs { get; set; }
 
-    public void OnGet()
+    public async Task OnGet()
     {
-        var laCode = TempData["LocalAuthorityCode"];
-        var localAuthority = _localAuthorityLookupService.GetLocalAuthorityFromCode((string)laCode);
+        var response = TempData["postcodeIoResponse"] as string;
+        if (response is null)
+            throw new Exception("Missig PostcodeIOResponse");
+
+        var postcodeIoResponse = JsonSerializer.Deserialize<PostcodeIOResponse>(response);
+
+        var laCode = (TempData["LocalAuthorityCode"] as string) ?? "";
+        double longitude = postcodeIoResponse?.Result?.Longitude ?? 0.0;
```

Comment:
What happens if we encounter a null when retrieving lat or long?
It would be best to not display a distance, or show something like N/A on the UI.

Does defaulting them to `0.0` end up showing the distance to 0:0?